lib/Twig/Node/Deprecated.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Represents a deprecated node.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class Twig_Node_Deprecated extends Twig_Node
+{
+    public function __construct(Twig_Node_Expression $expr, $lineno, $tag = null)
+    {
+        parent::__construct(array('expr' => $expr), array(), $lineno, $tag);
+    }
+
+    public function compile(Twig_Compiler $compiler)
+    {
+        $compiler
+            ->addDebugInfo($this)
+            ->write('@trigger_error(')
+            ->subcompile($this->getNode('expr'))
+            ->raw(".' (")
+            ->string($this->getTemplateName())
```

Comment:
that's broken. You should not use `->string` to put the double quotes around the template name inside the message. You should use it to render the PHP string (i.e. ` ("foo.twig" at line 2).`). Otherwise, that's a hacky usage of the API, and could break at any time (for instance if the compiler switches to using single-quoted strings when possible)

Reply:
Fixed, thanks again.